tests/functional/testAddons.js

```diff
@@ -164,22 +165,56 @@ describe('Addons', function () {
   describe('test message_subscription_expiring addon condition', async () => {
     const testCases = [
       ...Array.from(
-        { length: 7 },
-        // 1 to 7 days out from expiring.
-        (_, i) => [Date.now() + 1000 * 60 * 60 * 24 * (i + 1), true, `is ${i + 1} day(s) away`]
-      ),
+          {length: 7},
+          // 1 to 7 days out from expiring.
+          (_, i) =>
+              [true, () => Date.now() + 1000 * 60 * 60 * 24 * (i + 1), true,
+               `is ${i + 1} day(s) away (expiring)`]),
+      ...Array.from(
+          {length: 7},
+          // 1 to 7 days out from auto-renewing
+          (_, i) =>
+              [false, () => Date.now() + 1000 * 60 * 60 * 24 * (i + 1), false,
+               `is ${i + 1} day(s) away (renewing)`]),
       // Seven days out + five minutes  from expiring.
-      [Date.now() + 1000 * 60 * 60 * 24 * 7 + 1000 * 60 * 5, false, "is 7 days and five minutes away"],
+      [
+        true, () => Date.now() + 1000 * 60 * 60 * 24 * 7 + 1000 * 60 * 5, false,
```

Comment:
I had to use functions instead computing the date because on wasm this test takes more than 5 minutes and it breaks this and other dates